internal/pkg/manifest/workload.go

```diff
@@ -476,6 +480,82 @@ func UnmarshalWorkload(in []byte) (interface{}, error) {
 	}
 }
 
+// ContainerHealthCheck holds the configuration to determine if the service container is healthy.
+// See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-healthcheck.html
+type ContainerHealthCheck struct {
+	Command     []string       `yaml:"command"`
+	Interval    *time.Duration `yaml:"interval"`
+	Retries     *int           `yaml:"retries"`
+	Timeout     *time.Duration `yaml:"timeout"`
+	StartPeriod *time.Duration `yaml:"start_period"`
+}
+
+// newDefaultContainerHealthCheck returns container health check configuration
+// that's identical to a load balanced web service's defaults.
+func newDefaultContainerHealthCheck() *ContainerHealthCheck {
+	return &ContainerHealthCheck{
+		Command:     []string{"CMD-SHELL", "curl -f http://localhost/ || exit 1"},
+		Interval:    durationp(10 * time.Second),
+		Retries:     aws.Int(2),
+		Timeout:     durationp(5 * time.Second),
+		StartPeriod: durationp(0 * time.Second),
+	}
+}
+
+// apply overrides the healthcheck's fields if other has them set.
+func (hc *ContainerHealthCheck) apply(other *ContainerHealthCheck) {
+	if other.Command != nil {
+		hc.Command = other.Command
+	}
+	if other.Interval != nil {
+		hc.Interval = other.Interval
+	}
+	if other.Retries != nil {
+		hc.Retries = other.Retries
+	}
+	if other.Timeout != nil {
+		hc.Timeout = other.Timeout
+	}
+	if other.StartPeriod != nil {
+		hc.StartPeriod = other.StartPeriod
+	}
+}
+
+// applyIfNotSet changes the healthcheck's fields only if they were not set and the other healthcheck has them set.
+func (hc *ContainerHealthCheck) applyIfNotSet(other *ContainerHealthCheck) {
+	if hc.Command == nil && other.Command != nil {
+		hc.Command = other.Command
+	}
+	if hc.Interval == nil && other.Interval != nil {
+		hc.Interval = other.Interval
+	}
+	if hc.Retries == nil && other.Retries != nil {
+		hc.Retries = other.Retries
+	}
+	if hc.Timeout == nil && other.Timeout != nil {
+		hc.Timeout = other.Timeout
+	}
+	if hc.StartPeriod == nil && other.StartPeriod != nil {
+		hc.StartPeriod = other.StartPeriod
+	}
+}
+
+// HealthCheckOpts converts the image's healthcheck configuration into a format parsable by the templates pkg.
+func (i ImageWithPortAndHealthcheck) HealthCheckOpts() *ecs.HealthCheck {
+	if i.HealthCheck == nil {
+		return nil
+	}
+	// Make sure that unset fields in the healthcheck gets a default value.
+	i.HealthCheck.applyIfNotSet(newDefaultContainerHealthCheck())
```

Comment:
Is this the only change from the original code (in addition to moving them here)?

Reply:
yeah i think so.